### DISCUSSION TITLE
Is it possible to implement newer official or user-defined phase partitioning methods in NoahMP in LIS?

### DISCUSSION THREAD

**Discussion by ShangyongShi** [NASA-LIS/LISF#1632] [Q&A] [answered]
Dear developers,

I am working on comparing the different  phase partitioning methods in the Noah MP land surface model. I notice that in version 4.0.1, only three methods are available in LIS. The option to use WRF microphysics is muted according to the user manual. Futher, in later versions of Noah MP, a new wet-bulb scheme (Wang et al., 2019) is also added. So my questions are:

1. I learned from a previous discussion post in September that NCAR and NASA are working on adding Noah-MP-5.0 into LIS, and plan on having it available for use in a few months. I am wondering if an estimated publish time can be provided at this stage.

2. Is it possible to migrate just the phase partitioning module from newer Noah-MP versions to LIS?

3. I have located the code in /lis/surfacemodels/land/noahmp.4.0.1/phys/module_sf_noahmplsm_401.F90. Is it difficult to add other phase partitioning method by myself into the land surface model in the LIS? If more variables than surface temperature/RH need to be included, will added I/O interfere the compilation?

Thanks for your help!

**Comment by dmocko** [comment, marked as answer]
HI @ShangyongShi 

1. Yes, the plan to have Noah-MP-5.0 available in LIS is still a couple months away.  Most likely, it will be available to the public in January 2025.

2. and 3. If you want to modify that routine yourself for your version of LIS with Noah-MP-4.0.1, you are welcome to do so.

You said you already found the section of code where you would add the new method, but for others, it would be here:
https://github.com/NASA-LIS/LISF/blob/master/lis/surfacemodels/land/noahmp.4.0.1/phys/module_sf_noahmplsm_401.F90#L956-L1008

If you need to add more variables into this routine for your new method, then you will need to pass them in the call to this routine.
